miniscope_io/stream_daq.py

```diff
@@ -87,76 +77,143 @@ class MetadataHeader(BaseModel):
         header_fmt : MetadataHeaderFormat, optional
             Header format used to parse information from buffer header, by default `MetadataHeaderFormat()`.
-        header_len : int, optional
-            Length of header in (32-bit) words, by default 11.
-            This is useful when not all the variable/words in the header are defined in :class:`.MetadataHeaderFormat`.
-            The user is responsible to ensure that `header_len * 32` is larger than the largest bit position defined in :class:`.MetadataHeaderFormat` otherwise unexpected behavior might occur.
-        LSB : bool, optional
-            Whether the sourse is in "LSB" mode or not, by default True.
-            If `not LSB`, then the incoming bitstream is expected to be in Most Significant Bit first mode and data are transmitted in normal order.
-            If `LSB`, then the incoming bitstream is in the format that each 32-bit words are bit-wise reversed on its own.
-            Furthermore, the order of 32-bit words in the pixel data within the buffer is reversed (but the order of words in the header is preserved).
-            Note that this format does not correspond to the usual LSB-first convention and the parameter name is chosen for the lack of better words.
-        buffer_npix : Tuple[int], optional
-            A tuple defining how pixels within a single frame is split across multiple buffers, by default (20432, 20432, 20432, 20432, 10688).
-            Each number in the tuple represents how many pixels are contained in each buffer.
-            The length of the tuple represents the number of buffers a frame is split across.
-        pix_depth : int, optional
-            Bit-depth of each pixel, by default 8.
-        """
-        self.frame_width = frame_width
-        self.frame_height = frame_height
-        self.preamble = preamble
+        """        
+        self.logger = init_logger('streamDaq')
+
+
+        for key in config.model_json_schema()['properties'].keys():
+            config_dict = config.model_dump()
+
+            if key in config_dict:
+                if key == 'preamble':
+                    # Not ideal but needed because yaml can't handle hexadecimal files.
+                    self.preamble = bytes.fromhex(config_dict[key])
+                else:
+                    setattr(self, key, config_dict[key])
+            else:
+                self.logger.error(f"ERROR: {key} not found in config")
+
         self.header_fmt = header_fmt
-        self.header_len = header_len * 32
-        if LSB:
-            self.LSB = True
-        else:
-            self.LSB = False
-        self.buffer_npix = buffer_npix
-        assert frame_height * frame_width == sum(
-            self.buffer_npix
-        ), "Number of pixels defined by frame width and height must agree with total number of pixels in `buffer_npix`!"
+        px_per_frame = self.frame_width * self.frame_height
```

Comment:
Same thing here as above - make these computed fields in the config model since theyre all computed based on the config values. Field names in config model and this class are tightly coupled here, but also opaquely related bc of the `setattr` loop above (ie. if this or the model was changed, a static analysis tool wouldn't be able to tell)

Reply:
Done in https://github.com/Aharoni-Lab/miniscope-io/pull/21/commits/a8e1eaada4ddc9b51a7db0397866a1a4955b574a